libs/polkadot-api-provider/src/webb-provider/chain-query.ts

```diff
@@ -19,59 +20,66 @@ export class PolkadotChainQuery extends ChainQuery<WebbPolkadot> {
-        try {
-          tokenAccountData = await this.inner.api.query.tokens.accounts(
-            activeAccount.address,
-            assetId
-          );
-        } catch (e) {
-          // It is possible that we have connected to a chain that is not setup with orml tokens.
-          return '';
+  private getTokenBalanceByAssetId(assetId: string): Observable<string> {
+    return this.inner.newBlock.pipe(
+      switchMap(async () => {
+        const activeAccount = await this.inner.accounts.activeOrDefault;
+        if (activeAccount) {
+          // If the assetId is not 0, query the orml tokens
+          if (assetId !== '0') {
+            let tokenAccountData: OrmlTokensAccountData;
+
+            try {
+              tokenAccountData = await this.inner.api.query.tokens.accounts(
+                activeAccount.address,
+                assetId
+              );
+            } catch (e) {
+              // It is possible that we have connected to a chain that is not setup with orml tokens.
+              return '';
+            }
+
+            const json = tokenAccountData;
+
+            let tokenBalance: string = json.free.toString();
+
+            tokenBalance = tokenBalance.replaceAll(',', '');
+            const denominatedTokenBalance = Number(tokenBalance) / 10 ** 12;
+
+            return denominatedTokenBalance.toString();
+          } else {
+            const systemAccountData = await this.inner.api.query.system.account(
+              activeAccount.address
+            );
+
+            let tokenBalance: string = systemAccountData.data.free.toString();
+
+            tokenBalance = tokenBalance.replaceAll(',', '');
+            const denominatedTokenBalance = Number(tokenBalance) / 10 ** 12;
+
+            return denominatedTokenBalance.toString();
+          }
         }
 
-        const json = tokenAccountData;
-
-        let tokenBalance: string = json.free.toString();
-
-        tokenBalance = tokenBalance.replaceAll(',', '');
-        const denominatedTokenBalance = Number(tokenBalance) / 10 ** 12;
-
-        return denominatedTokenBalance.toString();
-      } else {
-        const systemAccountData = await this.inner.api.query.system.account(
-          activeAccount.address
-        );
-
-        let tokenBalance: string = systemAccountData.data.free.toString();
-
-        tokenBalance = tokenBalance.replaceAll(',', '');
-        const denominatedTokenBalance = Number(tokenBalance) / 10 ** 12;
-
-        return denominatedTokenBalance.toString();
-      }
-    }
-
-    return '';
+        return '';
+      })
+    );
   }
 
-  async tokenBalanceByCurrencyId(
+  tokenBalanceByCurrencyId(
     typedChainId: number,
     currency: CurrencyId
-  ): Promise<string> {
+  ): Observable<string> {
     const assetId =
       this.inner.config.currencies[currency].addresses.get(typedChainId);
-    return assetId ? this.getTokenBalanceByAssetId(assetId) : '';
+    if (!assetId) {
+      return throwError(`can't find asset`);
```

Comment:
For error messages we should refrain from using contractions (e.g. can't, shouldn't)